compiler/annotation/match_inference.rs

```diff
@@ -291,12 +295,14 @@ impl TypeInferenceGraph<'_> {
         for nested_graph in &mut self.nested_disjunctions {
             is_modified |= nested_graph.prune_vertices_from_self(&mut self.vertices);
         }
+        for nested_graph in &mut self.nested_optionals {
+            is_modified |= nested_graph.prune_vertices_from_constraints();
+        }
```

Comment:
Same as above. This shouldn't be here. Just in `prune_types_for_nested_negations_and_optionals`